src/YaccConstructor/Program.fs

```diff
@@ -91,34 +117,35 @@ let () =
     let generateSomething = ref true
 
     let printItems iName items deft =
-        fun _ ->
             generateSomething := false
             printfn "\nAvailable %s: " iName
             Seq.map (fun x -> x + (if Some x = deft then " (default)" else "")) items
             |> String.concat "\n    "
             |> fun x -> printf "    %s\n" x
 
-    let commandLineSpecs =
-        ["-f", ArgType.String (fun s -> feName := Some s), "Frontend name. Use -af to list available."
-         "-af", ArgType.Unit (printItems "frontends" addinFrontendNames !feName), "Available frontends"
-         "-g", ArgType.String 
-            (fun s -> 
-                match Array.toList (s.Split ' ') with
-                | name::[] -> generatorName := Some name; generatorParams := None
-                | name::parameters -> generatorName := Some name; generatorParams := Some (String.concat " " parameters)
-                | _ -> failwith "You need to specify generator name"
-            ), "Generator name. Use -ag to list available."
-         "-ag", ArgType.Unit (printItems "generators" addinGeneratorNames !generatorName), "Available generators"
-         "-c", ArgType.String (fun s -> conversions.Add s), "Conversion applied in order. Use -ac to list available."
-         "-ac", ArgType.Unit (printItems "conversions" addinConversionNames None), "Available conversions"
-         "-D", ArgType.String (fun s -> userDefs := !userDefs @ [s]), "User defined constants for YardFrontend lexer."
-         "-U", ArgType.String (fun s -> userDefs := List.filter ((<>) s) !userDefs), 
-                "Remove previously defined constants for YardFrontend lexer."
-         "-i", ArgType.String (fun s ->
-                                   testFile := System.IO.Path.GetFileName s |> Some
-                                   testsPath := System.IO.Path.GetDirectoryName s |> Some), "Input grammar"         
-         ] |> List.map (fun (shortcut, argtype, description) -> ArgInfo(shortcut, argtype, description))
-    ArgParser.Parse commandLineSpecs
+    let argv = System.Environment.GetCommandLineArgs()
+    let parser = ArgumentParser.Create<CLIArguments>(errorHandler = ProcessExiter())
+    let args = parser.Parse argv.[1..]
+    for res in args.GetAllResults() do
+        match res with 
+        | Frontend fe -> feName := Some fe
+        | AvailableFrontends -> printItems "frontends" addinFrontendNames !feName
+        | Generator g -> match Array.toList (g.Split ' ') with
+                         | name::[] -> 
+                            generatorName := Some name
```

Comment:
Unfortunately  (or not....) it is default behaviour of MSVS editor (if you are about indentation decreasing in nested match).  So, let we accept this variant. May be nested match should be placed on a new line. But it is not critical.